Bugfixes for search filter

pcraig3 opened the pull request:
This pull request kills two bugs:

- stops the form from submitting when a user presses "enter" while in the search filter
  - the search is entirely client-side so submitting the form from the search bar just refreshes the page and you lose your search terms
- resets the results after clicking the little 'x' that clears the search input field in some browsers


**Note**
I couldn't figure out how to write a test for the "enter" behaviour submitting/not submitting the form so at the moment there is no test for that behaviour. I'll ask within GDS if anyone has any recommendations.

## gifs 

| Before | After |
|--------|-------|
|  ![search-before](https://user-images.githubusercontent.com/2454380/28832764-48c29386-76d5-11e7-93b8-c55b5e583f4f.gif) | ![search-after](https://user-images.githubusercontent.com/2454380/28832771-4d936f34-76d5-11e7-8c4c-a70316b5d5fe.gif) |

olly commented on app/assets/javascripts/search.js:
```diff
@@ -77,14 +77,23 @@
       })
     }
 
+    var _preventSearchFilterSubmit = function () {
```
It might be more intention revealing, (and also cover a few more edge cases) if this binds to the `submit` event, and prevents that? I think that will achieve the same thing.

pcraig3 commented:
Cool, removed the input out of the form and now we don't need to listen for "enter" keypresses at all. Win/win.

(Note that I've added in some CSS here to fix a broken layout that I'm going to overhaul in a new pull request.)

olly commented:
👍
